string[] skill_names
float32[] progress_values
bool done